### PR TITLE
Fix py-libsonata

### PR DESCRIPTION
**Pull request by wizmer**

### CODE REVIEW

**Comment by wizmer** [comment]
https://github.com/BlueBrain/libsonata/issues/17

File: var/spack/repos/builtin/packages/py-libsonata/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    """
+    homepage = "https://github.com/BlueBrain/libsonata"
+    git = "https://github.com/BlueBrain/libsonata.git"
+
+    version('develop', git=git, submodules=True)
+
+    variant('mpi', default=False, description="Enable MPI backend")
+
+    depends_on('py-setuptools', type='build')
+    depends_on('libsonata@develop:', type='build')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-libsonata/package.py]: If we have an `mpi` variant, shouldn't this be build with/without `mpi`? @pramodk ?

**Comment by pramodk** [comment]
I am bit puzzled by this PR : are we going to maintain two separate packages?

If pybind version is an issue, can't we make `submodules=True` in libsonata itself?

**Comment by wizmer** [comment]
The other issue is that now libsonata is installable through `pip install` and no longer through a cmake call. So the py-libsonata recipe derives from PythonPackage and no longer from CMakePackage.

**Comment by pramodk** [comment]
@wizmer : the [jenkins plan](https://bbpcode.epfl.ch/ci/job/hpc.spack-deployment/configure) has list of GitHub users that can trigger plan. So we have to add GitHub username whenever there is new user.

**Comment by matz-e** [comment]
Also:

Ok to test

**Review by matz-e — CHANGES_REQUESTED** [review]
I think we have a stray thing that needs to be changed:
```
❯ rg "sonata\+python"
var/spack/repos/builtin/packages/py-bluepy/package.py
33:    depends_on('libsonata+python', type='run')
```

**Comment by pramodk** [comment]
> now libsonata is installable through pip install and no longer through a cmake call. 

@wizmer @arsenius7 : I assume you meant `also` pip (and CMake build is not deprecated)

Currently we have `libsonata` package in Spack and it's dependency of some other packages. For C++ projects depending on libsonata, CMake build is useful because we can easily change compilers, flags, static/shared builds through CMake options.

We can have this new package as an extra package (if this is useful for python packages) and fix libsonata  using `submodules=True`.

**Comment by wizmer** [comment]
@pramodk We keep cmake for C++ install of course but I would be in favor of deprecating the cmake way of installing the python package to reduce the maintenance cost.

**Comment by wizmer** [comment]
> I think we have a stray thing that needs to be changed:
> 
> ```
> ❯ rg "sonata\+python"
> var/spack/repos/builtin/packages/py-bluepy/package.py
> 33:    depends_on('libsonata+python', type='run')
> ```

Thanks !

**Comment by wizmer** [comment]
I think the Jenkins job as issues again.
https://bbpcode.epfl.ch/ci/job/hpc.spack-deployment/configure
Do I need to perform a specific action on my side (aka asking permissions) ?

**Comment by pramodk** [comment]
I have added your username in the jenkins plan so it should (have) trigger(ed). Lets see next time.

**Review by pramodk — APPROVED** [review]
@wizmer : just for info : note that this package won't be re-deployed  because spack will see that it's same version and dependencies (i.e. nothing has changed considering `spec` of the package). So would be better to tag versions if we start using libsonata in production.

**Review by matz-e — APPROVED** [review]

**Review by pramodk — APPROVED** [review]

File: var/spack/repos/builtin/packages/libsonata/package.py [hunk 2]
```diff
@@ -34,7 +35,7 @@ class Libsonata(CMakePackage):
 
     def cmake_args(self):
         result = [
-            '-DEXTLIB_FROM_SUBMODULES=OFF',
+            '-DEXTLIB_FROM_SUBMODULES=ON',
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/libsonata/package.py]: @wizmer : if we are using submodules now, could you check if all dependencies specified with `depends_on` are still required? (i.e. if py-pybind11 in being built as submodule, we can remove it from dependencies).

Comment (arsenius7) [reply var/spack/repos/builtin/packages/libsonata/package.py]: Btw, if other things being equal, `depends_on` is preferred, can't we bump `pybind11` version in Spack recipe instead?

Comment (pramodk) [reply var/spack/repos/builtin/packages/libsonata/package.py]: We have been using latest release, there is no upstream release with the fixes that we need (if I am not mistaken).

Comment (arsenius7) [reply var/spack/repos/builtin/packages/libsonata/package.py]: `pybind11@develop` or `pybind11@<commit>` until the necessary change become a part of the next `pybind11` release?

**Review by arsenius7 — APPROVED** [review]
LGTM as a way to revive `libsonata+python` for the moment.

File: var/spack/repos/builtin/packages/libsonata/package.py [hunk 3]
```diff
@@ -19,6 +19,8 @@ class Libsonata(CMakePackage):
     git = "https://github.com/BlueBrain/libsonata.git"
 
     version('develop', git=git, submodules=False)
+    version('v0.0.1', commit='6d7e3929c156812969d9237bd797bf6aba6e50ec', submodules=False)
```
Comment (arsenius7) [inline var/spack/repos/builtin/packages/libsonata/package.py:22]: sorry, I have overlooked `v` prefix in version here; will change it to `0.0.1` at #364.

**Pull request merged**